### ISSUE TITLE
[Q4] rancher2_app_v2 Ignoring Cluster Private Registry configuration and use only system default registry

### ISSUE THREAD

**Issue by matteogazzadi** [rancher/terraform-provider-rancher2#1187]
### Rancher Server Setup
- Rancher version: 2.7.5
- Global default registry configured on corporate private registry Azure ACR.

### Information about the Cluster
- Kubernetes version: v1.25.9-rancher2-1
- Cluster Type Downstream:
* Custom = Running a docker command on a node. Using RKE1

### User Information
- What is the role of the user logged in?  Admin

### Provider Information
- What is the version of the Rancher v2 Terraform Provider in use? 3.1.0
- What is the version of Terraform in use? 1.0.9

### Describe the bug

Our downstream infrastructure is designed to have a local container registry (Harbor) acting as proxy-cache towards the corporate container registry Azure Container Registry. Both are private registries. 
The idea is to reduce the overall required network bandwidth during commissioning and avoid to contact corporate registry when not needed.
 
The RKE1 downstream cluster is created using `rancher2_cluster`, and `private_registries` section is set as:
```
    private_registries {
      is_default = true
      url = "${var.harbor_dns_name}/${var.harbor_project_name}"
      user = "${harbor_robot_account.harbor_k8s_account.full_name}"
      password = "${random_password.harbor_k8s_account_password.result}"
    }
```

Harbor is managed still via Terraform and we set up a proxy-cache project towards our corporate Container Registry.

Cluster creation does work as expected, all containers are pulled from corporate private registry through local harbor.

Once Cluster is created, we want to install an app from rancher catalog using `rancher2_app_v2`.  The installation starts but the containers are pulled using the system default registry configured on rancher and not from the configured cluster private registry.
Of course this lead to ImagePullBackOff on any pod.

We tried to override the `global.cattle.system-default-registry` of the app but this does not work, See:  rancher/terraform-provider-rancher2#813 

We cannot set the system-default-registry in rancher as the one on downstream (completely no sense in general)

### To Reproduce

1. In Rancher configure a global default registry
2. Create a downstream cluster RKE1 using `rancher2_cluster` and configure a `private_registry` different from the Rancher global default registry
3. Install an app using `rancher2_app_v2`. 

### Actual Result

- App PODs Images are pulled from Rancher Global system default registry and not from the cluster private registry (the local Harbor).

### Expected Result
- App PODs Images are pulled from  the configured cluster private registry (the local Harbor).

### Screenshots
![image](https://github.com/rancher/terraform-provider-rancher2/assets/7020133/7608e43c-80bb-4bbe-bfd0-0803173f249f)

SURE-6737

**Comment by a-blender** [comment]
This is **_not expected behavior_** in the TF provider. As @rawmind0 has explained [in this comment](https://github.com/rancher/terraform-provider-rancher2/issues/813#issuecomment-989839192) and from my own investigation, you can't pass a system private registry value when creating a Rancher app v2 via Terraform. TF pulls the `system_default_registry` from Rancher v3 settings https://github.com/rancher/terraform-provider-rancher2/blob/baefd260c644dec490c77b192530327591432836/rancher2/resource_rancher2_app_v2.go#L52-L56

That being said, it's confusing why `system_default_registry` exists as a field in the TF provider in the first place if configurable input is not even used. TF _should_ just pull the needed Rancher v3 setting on the backend and pass it to the chart.

If we want to make this a configurable option for TF then we need to update the Rancher app v2 resource code to pass `system_default_registry` to the chart if configured.

**Comment by a-blender** [comment]
This is a Q4 enhancement targeted for TF for Rancher 2.8. If you need to configure a private registry to pull images from, please set the Rancher v3 global setting `system-default-registry` in the Rancher UI. Be aware that this will affect all downstream clusters.

**Comment by snasovich** [comment]
Since this primarily is about handling of `rancher2_app_v2` resource this should be looked into by Apps & Marketplace team. Assigning to @gunamata .

**Comment by l4zy0n3** [comment]
Validation Steps:
1. Downloaded the latest rancher2 beta provider containing the fixed code
2. Used dev_overrides to configure terraform to use the local beta provider
3. Ran the script and verified that the infrastructure was correct
4. Verified that the registry override for the cis-benchmark chart is present and working

provider version: rancher2 4.0.0-rc5_darwin_arm64

**Comment by snasovich** [comment]
@kkaempf , since this has been merged to `master` branch and is in Rancher TFP 4.0.0-rc5 milestone, this will be in the 4.0.0 release that we're tracking in `v2.8.0 - Terraform` milestone as Rancher TFP 4.0.0 will be the release corresponding to just-released Rancher v2.8.0. Updating the milestone back.
FYI @Jono-SUSE-Rancher
